### DISCUSSION TITLE
Light unit unevenness

### DISCUSSION THREAD

**Discussion by RMNB** [srozum/film_camera_tester#54] [Q&A] [answered]
Hi,

First, thank you for the awesome project! I put it together this weekend and the light response is almost perfectly linear using both my Nikon D3400 and Gossen Luna Pro Digital light meter (has reflective mode).

I have a question about whether my light unit is correct as I have some uneven brightness. On the sides of the diffuser I'm measuring 0.3-0.4 EV lower than the center with both the Nikon and the Gossen meter. Is this about what can be expected, or do I have an issue? Maybe wrong Fresnel lens was sent to me? Thanks for any help.

Some info:
Fresnel lens groves point away from led (ordered 50mm diameter, 40mm focal length, $1.55+shipping).
I replaced the driver Rcs resistor.
Voltage set to 13.45V to get 12EV.
My diffuser light transmission is 51% (not 59%), so I expected a slightly higher voltage.

![IMG_E0914](https://github.com/user-attachments/assets/4e26e18a-6e49-4cff-9f29-97401578880c)
![IMG_E0915](https://github.com/user-attachments/assets/58f9acbd-4f35-48b9-8495-d90d4d1a1160)

**Comment by Surreal62** [comment]
It appears you may have the Fresnel fitted the wrong way round.

**Reply by RMNB** [reply]
Instead of disassembling, I pulled out the light unit and felt the lens with my fingernails (gloved) and I could feel the groves from the diffuser side. I can try flipping it...

**Reply by RMNB** [reply]
But in general, sounds like what I have is not expected, correct?

**Reply by Surreal62** [reply]
It is unexpected.
The two I have are very even.
What exactly lens did you purchase?
I'll see if I can find the link for the ones I have.

**Reply by Surreal62** [reply]
I just found this on AliExpress: 
￡1.19 | 20 30 40 42 50 55 60 70 80 90 100 MM PMMA Round Fresnel Lens Focal Length 28 - 120mm Condenser Solar Energy Concentrator Lens
https://a.aliexpress.com/_EHgxENF

**Reply by RMNB** [reply]
That would be great. This was my order:
![image](https://github.com/user-attachments/assets/6de0deb0-3962-4c68-aedf-5f918246a2ea)

**Comment by srozum** [comment]
Hello @RMNB,

It almost certainly that you've got incorrect lens. But, first, check the orientation. First indicator that it's a wrong lens is voltage. 13.5V is too high. You should get even circle and correct 12EV at around 12.5V.

I pulled my order history and here's where I ordered my lens: https://www.aliexpress.us/item/2251832843866060.html

Also, other people posted what lens they got as 50mm diameter and 40mm focal length, and grooves pattern sometimes more coarse. For reference, here's photo of lens I used. Very fine circles.

![fresnel](https://github.com/user-attachments/assets/3a51b973-5377-41f2-b9e7-7e2854baaf18)

**Reply by RMNB** [reply]
I just ordered one of these and will try flipping the one I have and report back. I'll take a picture of it as well.

[https://www.aliexpress.us/item/3256803577283401.html?spm=a2g0o.order_list.order_list_main.14.622a1802cJYry4&gatewayAdapt=glo2usa](url)

**Comment by srozum** [comment]
@RMNB I see you skipped capacitors on main PCB, is Light Unit PCB also missing capacitors? Those are very important.

**Reply by RMNB** [reply]
Hi. I skipped those because I'm using a separate regulated 5V supply which has a 220µF capacitor on the output. The Assembly Instructions schematic has no capacitors for a regulated power supply, that's why I skipped it. Should I add them?

The light unit PCB has both capacitors in place.

**Reply by srozum** [reply]
Nah, should be fine. Arduino Nano, unlike Uno, has no capacitors on board to save space, so it's recommended to add your own if you are using its voltage regulator. But since you are providing 5V rail externally and capacitors are present then it should be fine.

Light Unit PCB must have one capacitor before buck converter and 330µF parallel to LED.

**Comment by RMNB** [comment]
Tried to flip the lens and the results were much worse so it was oriented correctly the first time.
![IMG_0935](https://github.com/user-attachments/assets/e56428ae-86b2-47c3-9c64-49559390c660)


Also adding a few additional photos of the light unit


![IMG_0936](https://github.com/user-attachments/assets/c7b34a15-aac5-42d6-8d33-0da979359cd2)
![IMG_0929](https://github.com/user-attachments/assets/dcbef68c-1ef2-4d65-acfa-7669c69b5065)
![IMG_0930](https://github.com/user-attachments/assets/0a497e52-0daa-4f26-959d-5b1aa98de51c)
![IMG_0928](https://github.com/user-attachments/assets/1f312654-7519-455d-ac08-d32fcc440591)
![IMG_0931](https://github.com/user-attachments/assets/638cffa4-9551-4f8f-be98-5b6b3a4c7582)
![IMG_0932](https://github.com/user-attachments/assets/3f4a50f4-fabe-4b07-9680-43e2c42ec818)
![IMG_0934](https://github.com/user-attachments/assets/c06a99dd-6c34-47bd-9fcd-f65b839eb2df)
![IMG_0926](https://github.com/user-attachments/assets/10f7cdfb-4ed8-4a3a-bac7-6c133c7cc6a7)

**Comment by Surreal62** [comment]
Can you check the focal length of your Fresnel?
It is difficult to tell from your images for certain if the Fresnel has as fine lines as on the ones I have, but they do look to be of a coarser pitch.

**Reply by RMNB** [reply]
Not sure if my method is correct, but looking at a computer screen through the lens, it’s blurry at 40mm and gets clearer at ~30mm and stays clear as I go down to 20mm.

**Reply by RobC62** [reply]
Try to focus with a pinpoint light (ceiling light will do) onto some paper - with the grooves away from the paper, the image should be sharp when the lens is 40mm away from the paper. All mine are dead on 40mm when the image is sharpest.

**Reply by RMNB** [reply]
With the lamp overhead shining down, I used the lens to concentrate light on a piece of paper. It was sharpest and most concentrated at exactly 40mm. (As if I was trying to start a fire) Shoot.

**Reply by RMNB** [reply]
But trying to read, I can't read the text when the lens is at 40mm away from the paper but it becomes sharp at 30mm.

**Reply by srozum** [reply]
As you can see, while focal length of fresnel lens is 40mm, the LED is placed closer than 40mm. Because LED has lens too, and combination of these two lens gives correct beam. Let's confirm size of your LED, thus it's lens size.

**Comment by srozum** [comment]
Yes, the fresnel lens looks good to me, and you confirmed focal length.

So back to original post where you stated 2 issues. 

First, light fall off at the corners. Ignore it. There is will be fall off but it's hard to measure without a spot meter. The main working area is center circle.

Second, 12EV at 13.45V. Although your original photo shows 12.91V, which is acceptable considering different diffuser.

Capacitors, Rcs, Buck converter all look good. Thanks to your additional photos.

So let's check LED. My question is what size it is? Because it's barely fits into opening. I measure mine is 6.86mm square.

**Reply by RMNB** [reply]
Hi Serhiy,

Measured the sides of the LED and its the same as yours. 
At 12.91V my cameras and light meter were a little too low but they all matched up at 13.45V.

Could it be that my diffuser is too opaque? There is another discussion about operating at lower voltage where someone with 69% transmission is setting 11.5V [https://github.com/srozum/film_camera_tester/discussions/38#discussioncomment-7809112](https://github.com/srozum/film_camera_tester/discussions/38#discussioncomment-7809112). Looks like you have about 60% transmission using 12.5 V, and I have 50% using ~13.5 V.

This is the one I bought: [https://www.ttplasticland.com/products/custom-white-translucent-2447-p95-matte-glossy-acrylic-sheet](https://www.ttplasticland.com/products/custom-white-translucent-2447-p95-matte-glossy-acrylic-sheet)



![IMG_0937](https://github.com/user-attachments/assets/843d153f-0981-4f22-ba04-a52d2dac8256)
![IMG_0938](https://github.com/user-attachments/assets/cca87f60-7d39-4cc4-90a5-e81f48f39d7e)

![IMG_0941](https://github.com/user-attachments/assets/7cb9b38c-6d4b-4143-8baa-673d8a4f1318)
![IMG_0942](https://github.com/user-attachments/assets/8e650a22-8a77-4e63-809d-f180b8f64764)
![IMG_0943](https://github.com/user-attachments/assets/dfcc5303-a71b-4ef5-9e09-39b02780f1bb)
![IMG_0944](https://github.com/user-attachments/assets/3c72f0ed-f4ae-47df-a738-e225e01a32d5)

**Reply by srozum** [reply]
Well, everything looks pretty good. I cannot think of anything else. If light levels are good at 13.5V, then ok, use it like this. You have separate 5V regulator, so Arduino is protected. Fan might run a bit fast or loud.

**Comment by RMNB** [comment]
Just want to give an update. I got one of the TAP plastics lighting white 69% light transmission sheets (they will send a 1/4 inch sample for $2+shipping if you request that at checkout) and I was able to drop the voltage to 12.5V. Not sure how someone else got 11.5V with that one unless they're using a 1/8 inch thick sheet.

I also tried another LED which I received with my initial order. It was sold with the same description but is a lighter yellow. That LED still worked fine, still just as linear, but it was measuring about 6000 Kelvin, so I went back to my original which is about 2800 K. Both LEDs have some light intensity drop-off in the corners.

Also got another Fresnel lens from Aliexpress and it worked exactly as my original.

One thing I'm noticing is that at the extremes of the EV range I'm losing linearity. At 16EV I measure about 16.2 and at 6 EV I have about 5.8. This is not a big deal and I'm happy to have a working light source. I just wish I had a "gold standard" I could calibrate to, especially since I have no idea which calibration constant my cameras and light meter use. I might try to borrow a Sekonic which states the calibration constant is 12.5.

Now I'm onto the sensors where I've had some interesting results as well. I'll post that in another thread.

**Reply by srozum** [reply]
Thanks for the update. I started worrying for you.

I am glad that new diffuser worked. 

As you may understand, diffuser and fresnel lens do not affect the curvature of a light output. If your old good LED measured differently, it may attribute to the measuring error. I measured not with light meter, but with illuminance meter.

I would say your Light Unit is good now.

**Comment by RMNB** [comment, marked as answer]
Want to give an update on a hack i tried to improve the light evenness. I tested a spare Fresnel lens by putting a dot (~3mm diameter) of White-Out in the center of the smooth side of the lens. The White-Out is easy to clean off if needed. Light evenness improved and now only the very corners have negligible falloff. Not sure there is any practical difference but I'm keeping the modification. I confirmed this with the same illuminance meter that Serhiy uses (found one cheap on ebay). BTW, for people who have an iPhone, I found the Lux Meter app works quite accurately if you place the phone lens against the screen. The app is supposed to be calibrated for each iphone model.

**Reply by srozum** [reply]
I’ve spent probably ten times more time on light source design than rest parts of the tester. Believe me it’s best you could get from given form factor. 

Because of the shape of the housing corners couldn’t be brighter. Don’t even try different colors, shapes or reflective paint inside - it will be worse in general.

**Reply by srozum** [reply]
Btw, illuminance is measured in Lux, luminance is measured in cd/m2. 
And those are different measuring devices
